src/ai/backend/manager/models/gql_models/container_registry_utils.py

```diff
@@ -0,0 +1,157 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Literal, Optional
+
+import aiohttp
+import aiohttp.client_exceptions
+import sqlalchemy as sa
+import yarl
+from sqlalchemy.ext.asyncio import AsyncSession as SASession
+from sqlalchemy.orm import load_only
+
+from ai.backend.logging import BraceStyleAdapter
+from ai.backend.manager.api.exceptions import (
+    ContainerRegistryNotFound,
+    GenericBadRequest,
+    InternalServerError,
+    NotImplementedAPI,
+    ObjectNotFound,
+)
+
+from ...container_registry import ContainerRegistryRow
+from ..association_container_registries_groups import (
+    AssociationContainerRegistriesGroupsRow,
+)
+from ..group import GroupRow
+from ..rbac import ProjectScope, ScopeType
+
+log = BraceStyleAdapter(logging.getLogger(__spec__.name))  # type: ignore
+
+
+async def handle_harbor_project_quota_operation(
+    operation_type: Literal["create", "read", "update", "delete"],
+    db_sess: SASession,
+    scope_id: ScopeType,
+    quota: Optional[int],
+) -> Optional[int]:
```

Comment:
Don't combine handler functions into `one function` that accept multiple operation_types.